[Tina Agarwal] iP

Thing1Thing2 opened the pull request:
# iP Pull Request
###### By Tina Agarwal/ Thing1Thing2
**In response to** :
> Week 4 [Fri, Aug 26th] - Project
iP: Use GFMD in the PR description

**How to use?**
- Clone repo `git clone`
- Run build

**Changes made:**
1. Change name to "Falcon"
2. Add Gradle
3. Rectify Issues found using Checkstyle
4. Add JUnit tests

:page_facing_up: **Duke class overview**:
```java
public class Duke {
    private Storage storage;
    private TaskList tasks;
    private UI ui;

    public Duke(String filePath) { ... }
    public void run() {...}
    public static void main(String[] args) {...}
}

```

:alarm_clock: **Increments pending as of [Week 4](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)**:
- [ ] Level-10
- [ ]  A-Checkstyle
- [ ]  A-Varargs
- [ ] A-Gradle
- [ ] A-JAR

Thing1Thing2 committed:
Add Input2, Input3, Input4 test cases

Thing1Thing2 committed:
Add Input5 test case

Thing1Thing2 committed:
Create DukeException class

Thing1Thing2 committed:
Create getDate() and getDescription()

Thing1Thing2 committed:
Define getDate() body

Thing1Thing2 committed:
Redefine handleTask bodies

Thing1Thing2 committed:
Define getDescription() body

Thing1Thing2 committed:
Marking error fixes

Thing1Thing2 committed:
Create enums

Thing1Thing2 committed:
Create getCommand()

Thing1Thing2 committed:
Move to switch statement

Thing1Thing2 committed:
Finish shift to switch

Thing1Thing2 committed:
Test changes made

Thing1Thing2 committed:
Add deleteTask

Thing1Thing2 committed:
Throw DukeException deleteTask

Thing1Thing2 committed:
Remove handleDelete

Thing1Thing2 committed:
Add missing break

Thing1Thing2 committed:
Create UserInputHistory class

Thing1Thing2 committed:
UserInputHistory class: move Duke methods here

Thing1Thing2 committed:
Shift to separate UserInputHistory class

Thing1Thing2 committed:
UserInputHistory class: complete shift

Thing1Thing2 committed:
Duke class: Remove import ArrayList

Thing1Thing2 committed:
UserInputHistory class: created writeToFile() and appendToFile()

Thing1Thing2 committed:
UserInputHistory class: remove writeToFile()

Thing1Thing2 committed:
UserInputHistory class: createIfDoesntExist()

Thing1Thing2 committed:
UserInputHistory class: handle file creation

Thing1Thing2 committed:
UserInputHistory class: create appendToFile()

Thing1Thing2 committed:
UserInputHistory class: store all task types

Thing1Thing2 committed:
UserInputHistory class: print file lines

Thing1Thing2 committed:
UserInputHistory class: Add updateUserInputHistory()

Thing1Thing2 committed:
UserInputHistory: Complete syncUserInputHistory()

Thing1Thing2 committed:
Handle IOExceptions

Thing1Thing2 committed:
UserInputHistory: create deleteLine()

Thing1Thing2 committed:
UserInputHistory class: Complete deleteLine()

Thing1Thing2 committed:
UserInputHistory class: getPath() correctly

Thing1Thing2 committed:
Refactor code with breaking changes

Thing1Thing2 committed:
Add storage support

Thing1Thing2 committed:
Merge branch 'branch-Level-7'

Thing1Thing2 committed:
Change type of date to LocalDate

Thing1Thing2 committed:
Create interface DateableTask

Thing1Thing2 committed:
Add isToday() and longDescription()

Thing1Thing2 committed:
Merge branch 'branch-Level-8'

Thing1Thing2 committed:
A-MoreOOP

Thing1Thing2 committed:
Add packages

Thing1Thing2 committed:
Finish writing methods

Thing1Thing2 committed:
More completion

Thing1Thing2 committed:
Fix utilities

Thing1Thing2 committed:
Modify mark command

Thing1Thing2 committed:
Resolving issues

Thing1Thing2 committed:
Fix issues

Thing1Thing2 committed:
Fix issues

Thing1Thing2 committed:
Fix issues

Thing1Thing2 committed:
Declare FindCommand and other methods

Thing1Thing2 committed:
Define methods

Thing1Thing2 committed:
Print find command output

Thing1Thing2 committed:
Merge branch 'branch-Level-9'

Thing1Thing2 committed:
Add build.gradle

Thing1Thing2 committed:
Add gradle

Thing1Thing2 committed:
Add checkstyle

Thing1Thing2 committed:
Modify build

Thing1Thing2 committed:
Creat test class

Thing1Thing2 committed:
Add stub classes

Thing1Thing2 committed:
Add more test classes

Thing1Thing2 committed:
Test StorageParser

Thing1Thing2 committed:
Remove unnecessary imports

Thing1Thing2 committed:
Fix style violations in Duke.java

Thing1Thing2 committed:
Fix style violations in AddDeadline.java

Thing1Thing2 committed:
Fix style violations in AddEvent.java and AddTask.java

Thing1Thing2 committed:
Fix style violations in command package

Thing1Thing2 committed:
Fix style violations in stroage package

Thing1Thing2 committed:
Fix style violations in task package

Thing1Thing2 committed:
Fix style violations in ui package

Thing1Thing2 committed:
Fix style violations in all files

Thing1Thing2 committed:
Fix style violations in all test files

Thing1Thing2 committed:
Merge pull request #1 from Thing1Thing2/branch-A-CodingStandard

Merge branch-A-CodingStandard into master

Thing1Thing2 committed:
Remove old test folder

Thing1Thing2 committed:
Merge branch 'add-gradle-support'

Thing1Thing2 committed:
Add to build.gradle

Thing1Thing2 committed:
Add to build.gradle

Thing1Thing2 committed:
Modify ui methods

Thing1Thing2 committed:
Add/Remove files

Thing1Thing2 committed:
Add images

Thing1Thing2 committed:
Add fxml files

Thing1Thing2 committed:
Modify Main and MainWindow

Thing1Thing2 committed:
Resolve file storage issue

JonathanWiguna commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,55 @@
+    /**
+     * Constructor.
+     * @param filePath Path where disk file should be created.
+     */
```
I think there should be a line between the description and the parameter section of your JavaDocs.

JonathanWiguna commented on src/main/java/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,40 @@
+public class AddDeadlineCommand extends Command {
+
+    /**
+     * Runs when deadline is to be added.
+     * @param taskList TaskList to append Deadline to.
```
Same here

Thing1Thing2 committed:
Finish basic GUI addition

Thing1Thing2 committed:
Merge branch 'branch-Level-10'

JonathanWiguna commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,55 @@
+import command.Command;
```
Shouldn't Duke be in a package as well?

JonathanWiguna commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,86 @@
+ */
+public class Storage {
+    private Path path;
+    private final StorageReader storageReader;
+    private final StorageWriter storageWriter;
```
Since these are constants, I think they should be in all uppercase with underscore to separate the words

Thing1Thing2 replied:
Ah yes, thank you for mentioning that :D

JonathanWiguna commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,86 @@
+
+    //Define all StorageWriter methods
+
+    public boolean appendLine(String s) {
+        return storageWriter.appendLine(s);
+    }
+
+    public boolean deleteLine(int index) {
+        return storageWriter.deleteLine(index);
+    }
+
+    public boolean changeLine(int index, String newString) {
+        return storageWriter.changeLine(index, newString);
+    }
```
Boolean methods should sound like booleans right?

Thing1Thing2 replied:
True, I got a little confused since changeLine causes change as well as returning true if the change was successful.

Thing1Thing2 committed:
Track data/tasks.txt

JonathanWiguna reviewed (commented):
Looks good! Just some small changes regarding coding practice requested.

Thing1Thing2 committed:
Storage: fix filePath

Thing1Thing2 committed:
Checkstyle

Thing1Thing2 committed:
Add shadowJar

Thing1Thing2 committed:
Fix file path issue

Thing1Thing2 committed:
Fix file path issue

Thing1Thing2 committed:
Implement PR comments

Thing1Thing2 committed:
Add asserts

Thing1Thing2 committed:
Fix javadoc

Thing1Thing2 committed:
StorageParser class: remove magic values

Thing1Thing2 committed:
Improve code quality

Thing1Thing2 committed:
Make 'happy path' more prominent

Thing1Thing2 committed:
Add obvious condition names

Thing1Thing2 committed:
StorageReader: getAllLines() access private

Thing1Thing2 committed:
Commit files

Thing1Thing2 committed:
Run checkstyle

Thing1Thing2 committed:
Check line lengths and variable names

Thing1Thing2 committed:
Storage class: rename boolean methods

Thing1Thing2 committed:
Storage class: rename boolean methods

Thing1Thing2 committed:
Storage class: rename boolean methods

Thing1Thing2 committed:
Duke class: remove unused imports

Thing1Thing2 committed:
Fix more issues

Thing1Thing2 committed:
Add new line in JavaDoc comments

Thing1Thing2 committed:
Make class variables private

Thing1Thing2 committed:
Remove redundant exit() command

Thing1Thing2 committed:
Fix code style

Thing1Thing2 committed:
Fix code style

Thing1Thing2 committed:
Run checkstyle

Thing1Thing2 committed:
Add stream support to TaskList

Thing1Thing2 committed:
Sync arraylist using streams

Thing1Thing2 committed:
Merge pull request #2 from Thing1Thing2/branch-asserts

Branch asserts

Thing1Thing2 committed:
Resolve conflicts in remote repo.

After remote merge of branch-assets and master, conflicts are present.

Pull remote master branch and resolve locally.
Push changes to remote master.

Thing1Thing2 committed:
Merge branch 'master' into A-CodeQuality

Thing1Thing2 committed:
Merge pull request #3 from Thing1Thing2/A-CodeQuality

A-CodeQuality

Thing1Thing2 committed:
Ensure program works after merge

Thing1Thing2 committed:
Merge branch 'master' into A-Streams

Thing1Thing2 committed:
Merge pull request #4 from Thing1Thing2/A-Streams

A-Streams

Thing1Thing2 committed:
Resolve conflicts after A-Streams merge

Thing1Thing2 committed:
Launcher class: simplify findTask()

Thing1Thing2 committed:
Merge pull request #5 from Thing1Thing2/branch-A-CodeQuality

Launcher class: simplify findTask()

Thing1Thing2 committed:
Add .yml for GitHub Actions CI

Thing1Thing2 committed:
Merge branch 'master' of https://github.com/Thing1Thing2/ip

Thing1Thing2 committed:
Add todo command alias support

Thing1Thing2 committed:
Add command aliases: add shortcuts

Adding command is rigid and memory-intensive.
User should be able to type intuitively.

Add shortcuts to existing commands.
Perform simple input validation.

Note: Use Regex, and LocalDateTime class as much as possible.

Thing1Thing2 committed:
Add support for aliases

Thing1Thing2 committed:
Reduce logical complexity

Thing1Thing2 committed:
Add support for aliases

Thing1Thing2 committed:
Add case insensitivity to alias checks

Thing1Thing2 committed:
Store actual command used in associated class

Thing1Thing2 committed:
Add case insensitivity to alias checks

Thing1Thing2 committed:
Store sliced user input in command

Thing1Thing2 committed:
Handle description

Thing1Thing2 committed:
Handle description

Thing1Thing2 committed:
Handle help command

Thing1Thing2 committed:
Improve code quality

Thing1Thing2 committed:
Improve code quality

Thing1Thing2 committed:
Change to matches()

Thing1Thing2 committed:
Build alias hashmap

Thing1Thing2 committed:
Build alias hashmap

Thing1Thing2 committed:
Improve code quality

Thing1Thing2 committed:
Run checkstyle

Thing1Thing2 committed:
Merge pull request #6 from Thing1Thing2/branch-C-FriendlierSyntax

lgtm, add changes suggested as-and-when necessary.l

Thing1Thing2 committed:
Narrow down issue

Thing1Thing2 committed:
Add missing regex

Thing1Thing2 committed:
Remove delete command parse issue

Thing1Thing2 committed:
Remove event command parse issue

Thing1Thing2 committed:
Remove repeated addTask commands

Thing1Thing2 committed:
Merge branch 'branch-Remove-Rawtype'

Thing1Thing2 committed:
Add colours

Thing1Thing2 committed:
MainWindow: ScrollPanel VBar policy always

Thing1Thing2 committed:
MainWindow: Fix anchorpane position

Thing1Thing2 committed:
MainWindow: Fix anchorpane position

Thing1Thing2 committed:
DialogBox: change resizability option

Thing1Thing2 committed:
DialogBox: correct label text size

Thing1Thing2 committed:
Create separate DialogBox types

Thing1Thing2 committed:
Copy DialogBox code to separate DialogBox types

Thing1Thing2 committed:
Create DialogBox files

Thing1Thing2 committed:
Copy DialogBox code to files

Thing1Thing2 committed:
Change DialogBox components ids

Thing1Thing2 committed:
Add new dialog boxes to MainWindow

Thing1Thing2 committed:
Fix field names in DialogBoxes

Thing1Thing2 committed:
Fix DialogBoxFalcon path issue

Thing1Thing2 committed:
Merge branch 'branch-A-BetterGui'

Thing1Thing2 committed:
Add 'task' alias

Thing1Thing2 committed:
Fix code issues

Thing1Thing2 committed:
Add Ui.png

Thing1Thing2 committed:
Explain basic commands

Thing1Thing2 committed:
Update README.md

Thing1Thing2 committed:
Reduce intialiseCommandAliases() size

Thing1Thing2 committed:
Reduce intialiseCommandAliases() size

Thing1Thing2 committed:
Parser: remove assert

Thing1Thing2 committed:
StorageParser: simplify fileLindToTask()

Thing1Thing2 committed:
StorageParser: improve slap

Thing1Thing2 committed:
StorageParser: improve slap

Thing1Thing2 committed:
StorageParser: improve slap

Thing1Thing2 committed:
StorageParser: improve slap

Thing1Thing2 committed:
Storage: improve slap

Thing1Thing2 committed:
StorageReader: improve slap

Thing1Thing2 committed:
StorageWriter: improve slap

Thing1Thing2 committed:
Fix logic of AddDeadlineCommand

Thing1Thing2 committed:
Fix logic of AddEventCommand

Thing1Thing2 committed:
Fix logic of AddTaskCommand

Thing1Thing2 committed:
Fix logic of CheckIsTodayCommand

Thing1Thing2 committed:
Fix logic of DeleteTaskCommand

Thing1Thing2 committed:
Fix logic of ExitCommand

Thing1Thing2 committed:
Fix logic of FindCommand

Thing1Thing2 committed:
Fix logic of GetLongDescriptionCommand

Thing1Thing2 committed:
Fix logic of HelpCommand

Thing1Thing2 committed:
Fix logic and slap of ListCommand

Thing1Thing2 committed:
Fix logic of MarkCommand

Thing1Thing2 committed:
Fix logic of UnmarkCommand

Thing1Thing2 committed:
Parser: fix parse() cases

Thing1Thing2 committed:
Parser: fix parse() cases

Thing1Thing2 committed:
Parser: introduce new method definitions

Thing1Thing2 committed:
Parser: create method definitions

Thing1Thing2 committed:
Parser: create method definitions

Thing1Thing2 committed:
Parser: build method bodies

Thing1Thing2 committed:
Parser: build method bodies

Thing1Thing2 committed:
Parser: build method bodies

Thing1Thing2 committed:
Parser: build getEventDate() body

Thing1Thing2 committed:
Parser: build deadline creating method bodies'

Thing1Thing2 committed:
Clear test file

Thing1Thing2 committed:
Parser: remove unused import

Thing1Thing2 committed:
AddEventCommand: remove unused import

Thing1Thing2 committed:
AddTaskCommand: remove unused import

Thing1Thing2 committed:
Implement checkstyle suggestions

Thing1Thing2 committed:
Implement checkstyle suggestions

Thing1Thing2 committed:
Implement checkstyle suggestions

Thing1Thing2 committed:
Implement checkstyle suggestions

Thing1Thing2 committed:
Move methods from Parser to StorageParser

Thing1Thing2 committed:
Implement checkstyle suggestions

Thing1Thing2 committed:
Implement checkstyle suggestions

Thing1Thing2 committed:
DialogBox: make lines shorter

Thing1Thing2 committed:
TaskList: improve findAllMatchingTasks() slap

Thing1Thing2 committed:
Parser: make matching more concise

Thing1Thing2 committed:
Add..Command: extract storage condition

Thing1Thing2 committed:
Add..Command: extract storage condition

Thing1Thing2 committed:
Avoid complicated expressions

Thing1Thing2 committed:
Merge branch 'master' of https://github.com/Thing1Thing2/ip

Thing1Thing2 committed:
Provide UG overview

Thing1Thing2 committed:
UG Introduction

Thing1Thing2 committed:
Update UG Quick Start section

Thing1Thing2 committed:
Update features list

Thing1Thing2 committed:
Update feature list

Thing1Thing2 committed:
Update feature list

Thing1Thing2 committed:
Finish up UG

Thing1Thing2 committed:
Delete regex error

Thing1Thing2 committed:
Merge branch 'master' of https://github.com/Thing1Thing2/ip

Thing1Thing2 committed:
Update tasks.txt

Thing1Thing2 committed:
Make window scroll down automatically

Thing1Thing2 committed:
Update README.md

Thing1Thing2 committed:
Update README.md

Thing1Thing2 committed:
Update README.md

Thing1Thing2 committed:
Add gradle wrapper jar

Thing1Thing2 committed:
Add test cases with exception handling

Thing1Thing2 committed:
Changing permission of gradlew

Thing1Thing2 committed:
Create gradle-wrapper.properties

Thing1Thing2 committed:
Reset wrapper

Thing1Thing2 committed:
Merge branch 'requirement-testing-and-exceptions'

Thing1Thing2 committed:
Merge branch 'master' of https://github.com/Thing1Thing2/ip

Thing1Thing2 committed:
Remove failing test case

Thing1Thing2 committed:
Remove unnecessary import

Thing1Thing2 committed:
Remove runTest from yaml
